tests/ops/test_qubit_ops.py

```diff
@@ -865,6 +867,152 @@ def test_controlled_phase_shift_decomp(self, phi):
+
+        pytest.importorskip("autograd")
+        dev = qml.device('default.qubit.autograd', wires=2)
+        state = np.array([0, -1 / np.sqrt(2), 1 / np.sqrt(2), 0])
+
+        @qml.qnode(dev)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            excitation(phi, wires=[0, 1])
+            return qml.state()
+        assert np.allclose(state, circuit(np.pi / 2))
+
+    @pytest.mark.parametrize(("excitation", "phi"), [(qml.SingleExcitation, -0.1),
+                                                     (qml.SingleExcitationPlus, 0.2),
+                                                     (qml.SingleExcitationMinus, np.pi/4)])
+    def test_autograd_grad(self, excitation, phi):
+        """Tests that gradients are computed correctly using the
+        autograd interface"""
+
+        pytest.importorskip("autograd")
+        dev = qml.device('default.qubit.autograd', wires=2)
+
+        @qml.qnode(dev)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            excitation(phi, wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+        assert np.allclose(qml.grad(circuit)(phi), np.sin(phi))
+
+    @pytest.mark.parametrize("diff_method", ["parameter-shift", "backprop"])
+    @pytest.mark.parametrize(("excitation", "phi"), [(qml.SingleExcitation, -0.1),
+                                                     (qml.SingleExcitationPlus, 0.2),
+                                                     (qml.SingleExcitationMinus, np.pi / 4)])
+    def test_tf(self, excitation, phi, diff_method):
+        """Tests that gradients and operations are computed correctly using the
+        tensorflow interface"""
+
+        tf = pytest.importorskip("tensorflow")
+        dev = qml.device('default.qubit', wires=2)
+
+        @qml.qnode(dev, interface="tf", diff_method=diff_method)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            excitation(phi, wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+            
+        phi_t = tf.Variable(phi, dtype=tf.float64)
+        with tf.GradientTape() as tape:
+            res = circuit(phi_t)
+
+        grad = tape.gradient(res, phi_t)
+        assert np.allclose(grad, np.sin(phi))
+
+    @pytest.mark.parametrize("diff_method", ["parameter-shift", "backprop"])
+    @pytest.mark.parametrize(("excitation", "phi"), [(qml.SingleExcitation, -0.1),
+                                                     (qml.SingleExcitationPlus, 0.2),
+                                                     (qml.SingleExcitationMinus, np.pi / 4)])
+    def test_jax(self, excitation, phi, diff_method):
+        """Tests that gradients and operations are computed correctly using the
+        jax interface"""
+        
+        if diff_method=="parameter-shift":
+            pytest.skip("JAX support for the parameter-shift method is still TBD")
+
+        jax = pytest.importorskip("jax")
+        from jax import numpy as jnp
+
+        dev = qml.device('default.qubit', wires=2)
```

Comment:
Just for the sake of making the test more explicit and readable
```suggestion
        dev = qml.device('default.qubit.jax', wires=2)
```